### PR TITLE
Improve contributor documentation.

### PR DESCRIPTION
**Pull request by nicksantamaria**
#### What does this PR do?

* Adds some improved contributor documentation
* Adds a ready-to-go development environment 

#### How should this be manually tested?

1. Check out this branch
1. Run a few commands to test.

```bash
docker-compose run --rm dev make lint
docker-compose run --rm dev make test
docker-compose run --rm dev make build
```

#### What are the relevant tickets?

https://redmine.previousnext.com.au/issues/47648

### CODE REVIEW

**Comment by nickschuch** [comment]
Vagrant, or Docker? ;)

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+  config.vm.box = "bento/ubuntu-#{UBUNTUVERSION}"
+  config.vm.hostname = "terraform"
+
+  config.vm.provision "prepare-shell", type: "shell", inline: "sudo sed -i '/tty/!s/mesg n/tty -s \\&\\& mesg n/' /root/.profile", privileged: false
+  config.vm.provision "initial-setup", type: "shell", inline: $script
+  config.vm.synced_folder '.', '/opt/gopath/src/github.com/previousnext/m8s'
+
+  config.vm.provider "docker" do |v, override|
```
Comment (nicksantamaria) [inline Vagrantfile]: It uses the docker driver by default. If people want to use a different vm engine, just a matter of `vagrant up --provider=virtualbox`.

**Comment by nicksantamaria** [comment]
By using the existing Vagrant solution from Terraform it saved a bunch of time. Also the stakes are not that high - this just a convenience for people having a dabble at m8s dev. I dont think it will suit for serious development as there is no support for debugging.

**Comment by nicksantamaria** [comment]
As discussed in today's meeting I'll convert the Vagrantfile to docker-compose.

**Pull request merged**